Gems/ROS2/Code/Source/Manipulator/JointPublisherComponent.cpp

```diff
@@ -0,0 +1,118 @@
+#include <ROS2/Manipulator/JointPublisherComponent.h>
+#include <AzCore/Serialization/EditContext.h>
+#include "ROS2/ROS2Bus.h"
+#include <PhysX/Joint/PhysXJointRequestsBus.h>
+#include <Source/HingeJointComponent.h>
+#include <AzToolsFramework/Entity/EditorEntityHelpers.h>
+
+namespace ROS2
+{
+    void JointPublisherComponent::Activate()
+    {
+        AZ::TickBus::Handler::BusConnect();
+        auto ros2Node = ROS2::ROS2Interface::Get()->GetNode();
+        AZStd::string jointStatesTopic = "joint_states";
+        m_jointstatePublisher = ros2Node->create_publisher<sensor_msgs::msg::JointState>(jointStatesTopic.data(), rclcpp::QoS(1));        // TODO: add QoS instead of "1"
```

Comment:
Looks to me like the publisher might need to be destroyed during Deactivate()?

Reply:
I agree